timemachine/fe/system.py

```diff
@@ -137,41 +126,94 @@ def U_fn(x):
 
         return U_fn
 
+    @abstractmethod
+    def get_U_fns(self):
+        raise NotImplementedError()
+
+
+@dataclass  # mcwitt: Generic can be removed in python 3.12
+class HostSystem(Generic[_HarmonicAngle], AbstractSystem):
```

Comment:
nit/future: I wonder if we can just get rid of the current `HarmonicAngle` and rename `HarmonicAngleStable` (since IIRC the former is a special case of the latter with `eps=0`). Then we could remove this polymorphism.

Reply:
yes, and lets defer it to a future case (since it will enlarge this PR considerably)